src/corehost/cli/test/nativehost/nativehost.cpp

```diff
@@ -50,13 +50,13 @@ int main(const int argc, const pal::char_t *argv[])
         pal::string_t fxr_path;
         size_t len = fxr_path.size();
         int res = get_hostfxr_path(nullptr, &len, assembly_path);
-        if (res == StatusCode::HostApiBufferTooSmall)
+        if (static_cast<StatusCode>(res) == StatusCode::HostApiBufferTooSmall)
```

Comment:
If `get_hostfxr_path` (and similar methods) returned a `StatusCode`, we can avoid these repeated `static_cast<StatusCode>`s. But I am not sure if that would be a API/ABI compatible change.

Reply:
We don't use enums in the exported functions - I guess for simplicity. To be honest, in this case I think the compiler is way too strict. If I'm comparing int to an enum literal, it shows intent pretty well I think. Having to add a static cast just feels wrong.

Reply:
Would you like me to revert this change?

Reply:
Let's leave it in... I'll take a look if we can somehow simplify this, but we can live with it as is.